.vscodeignore

```diff
@@ -1,10 +1,15 @@
 .vscode/**
 .vscode-test/**
-src/**/*.ts
+src/**
 .gitignore
 .yarnrc
 vsc-extension-quickstart.md
 **/tsconfig.json
 **/.eslintrc.json
 **/*.map
 **/*.ts
+CODEOWNERS
+RELEASE.md
+coverage/
+.nyc_output/
+.github
```

Comment:
We need to do this with folders like `out/` and `node_modules/` too. There should be a very lightweight, file-count-wise, package at the end of the line, because everything is meant to be bundled into one file.

Reply:
If I remove node_modules, the generated extension does not work. Also, the out is needed because that is where our extension is compiled to, which is our entrypoint. I dont think we can discard these two dirs.

Reply:
Yeah I'm basically fiddling around in my environment, seeing similar things.

There's a gap of documentation about some of the entries in your standard `package.json` file vs. what `esbuild` is / might be expecting. For example, I switched locally to have esbuild use an `--outfile` value of `dist/main.js` just to see how it went. And while it generates an all-in-one `dist/main.js`, it still complains at runtime about trying to use `out/extension.js`, which is the `main` entry point in `package.json`.

The dearth of comprehensive documentation is...frustrating. I'm going to keep poking around.

Reply:
if you just change `--outfile` to `out/extension.js` it should work though so that it matches package.json's `main` value?

Reply:
Yeah but I didn't like that because that's where we compile everything out too, and so you have to do a bunch of filtering to not include a swathe of JS files down there.

Locally, I ended up specifying `--outfile` as `dist/main.js`, and made that my `main` target in package.json too, and that got past that particular hurdle (and onto another one, but that's another story).